### ISSUE TITLE
`wp_die()` doesn't end script anymore

### ISSUE THREAD

**Issue by Screenfeed** [szepeviktor/phpstan-wordpress#203]
Hello,

Sorry to bring bad news. Unless I'm missing something, it seems that something went wrong with #201.

```php
/**
 * @phpstan-return never
 */
function foo() {
	wp_die();
}
```

This ends with the following message:
>Function foo() should always throw an exception or terminate script execution but doesn't do that.

**Comment by szepeviktor** [comment]
@IanDelMar Please hotfix this 🔥

**Comment by szepeviktor** [comment]
Seems like `@return never` is does not mean termination.

**Comment by szepeviktor** [comment]
@IanDelMar Wait! `wp_die` got no `@return never` in #110!

**Comment by johnbillion** [comment]
Looks like a test is needed for `assertType('*NEVER*', wp_die(''));` too (no second parameter).

**Comment by IanDelMar** [comment]
Was at work. Will have a look at it now.

**Comment by IanDelMar** [comment]
I added tests for `wp_die()` and `wp_die('')`. Both tests pass, because the extension returns "never" if the number of arguments is less than 3:
https://github.com/szepeviktor/phpstan-wordpress/blob/78db560e7989b50b05bec50ee5d862b41892654a/src/WpDieDynamicFunctionReturnTypeExtension.php#L31-L34

**Comment by Screenfeed** [comment]
@szepeviktor Sorry I should have mentioned that in the OP.

```
php-stubs/wordpress-stubs v6.3.2
szepeviktor/phpstan-wordpress v1.3.1
```

**Comment by szepeviktor** [comment]
> Does PHPStan know about the dynamic return type of a function when that function is called by another function?

Yes.

**Comment by szepeviktor** [comment]
`\PHPStan\dumpType(wp_die());` says NEVER.

@IanDelMar Adding back `wp_die` to `earlyTerminatingFunctionCalls` resolves the problem.

**Comment by szepeviktor** [comment]
@ondrejmirtes Could you help us?
Is there a programmatic way to mark a function early terminating?

**Comment by IanDelMar** [comment]
Look at this:
```php
/**
 * @phpstan-return never
 */
function foo() {
	\PHPStan\dumpType(wp_die());
	wp_die();
}
\PHPStan\dumpType(foo());
```
```
:51    Dumped type: *NEVER*
:52    Function foo() should always throw an exception or terminate script execution but doesn't do that.
:54    Dumped type: never
```

**Comment by szepeviktor** [comment]
# 😲

**Comment by szepeviktor** [comment]
Yess please!

**Comment by IanDelMar** [comment]
I was wondering why it's `'*NEVER*'` not just `'never'`, but was not aware that there is another never type (`NonAcceptingNeverType`) and thought PHPSTan will know that `'*NEVER*'` is `'never'`.

**Comment by johnbillion** [comment]
`WordPress will *NEVER* die()`

**Comment by ondrejmirtes** [comment]
Yeah, use `new NeverType(true)` or `new NonAcceptingNeverType()`. The NeverType needs to be explicit: https://github.com/phpstan/phpstan-src/blob/485922f5c2734af5af15230250340e6c1cc76451/src/Analyser/NodeScopeResolver.php#L1846-L1848

**Comment by szepeviktor** [comment]
Thank you.

**Comment by szepeviktor** [comment]
@Screenfeed This release is for you.
https://github.com/szepeviktor/phpstan-wordpress/releases/tag/v1.3.2

**Comment by manooweb** [comment]
> @Screenfeed This release is for you. https://github.com/szepeviktor/phpstan-wordpress/releases/tag/v1.3.2

Yes! It's ok for us 👍 Thank you so much 😉

**Comment by Screenfeed** [comment]
@szepeviktor Perfect. Thanks a lot to everyone! ❤️
